earthmover/nodes/source.py

```diff
@@ -331,14 +332,68 @@ def _get_read_lambda(self, file_type: str, sep: Optional[str] = None):
         """
         # Define any other helpers that will be used below.
         def __get_skiprows(config: 'YamlMapping'):
-            """ Retrieve or set default for header_rows value for CSV reads. """
+            """ Determine (from `header_rows`) how many rows to skip when reading CSV/TSV/Excel files. """
             _header_rows = config.get('header_rows', 1)
-            return int(_header_rows) - 1  # If header_rows = 1, skip none.
+            if type(_header_rows) is list:
+                return max(_header_rows)
+            elif type(_header_rows) is int or type(_header_rows) is str:
+                return int(_header_rows) - 1  # If header_rows = 1, skip none.
+            else:
+                self.error_handler.throw(
+                    "source `header_rows` must be an integer (the row number to use as the header) or list (of row numbers that constitute a multi-line header)"
+                )
+
+        def __get_flattened_columns(file, config: 'YamlMapping'):
+            """ Flatten and (potentially) fill multi-level (potentially sparse) headers for CSV, TSV, and Excel reads. """
+            _header_rows = config.get('header_rows', 1)
+            _fill_sparse_headers = config.get('fill_sparse_headers', False)
+            if type(_header_rows) is list:
+                pass
+            elif type(_header_rows) is int or type(_header_rows) is str:
+                _header_rows = [ int(_header_rows) - 1 ]
+            else:
+                self.error_handler.throw(
+                    "source `header_rows` must be an integer (the row number to use as the header) or list (of row numbers that constitute a multi-line header)"
+                )
+            # read the first few rows of the file to load the (potentially multi-line, potentially sparse) header
+            if file_type == 'csv' or file_type == 'tsv':
+                # (use `pd.read_csv()` - not `dd.read_csv()` - because dask doesn't support a [list] `header`)
+                df = pd.read_csv(file, sep=sep, dtype=str, encoding="utf8", keep_default_na=False, header=_header_rows, nrows=max(_header_rows)+1)
+            elif file_type == 'excel':
+                df = pd.read_excel(file, sheet_name=config.get("sheet", 0), keep_default_na=False, header=_header_rows, nrows=max(_header_rows)+1)
+            else:
+                self.error_handler.throw(
+                    "internal error: `__get_flattened_columns()` may only be called for csv, tsv, or excel `file_type`s."
+                )
+            if df.columns.nlevels == 1:
+                flattened_columns = df.columns
+            else:
+                # flatten multi-line header to tuples
+                flattened_columns = list(map(list, zip(*[            # 4. pivot cols x levels to levels x cols
```

Comment:
I think that for this list comprehension and the one below, it would be good to include how a simple example multi-line header would be transformed at each step.